external-crates/move/move-core/types/src/language_storage.rs

```diff
@@ -181,6 +199,27 @@ impl StructTag {
         )
     }
 
+    /// Return the canonical string representation of the struct, including the prefix '0x'
+    pub fn to_canonical_string_with_prefix(&self) -> String {
+        let mut generics = String::new();
+        if let Some(first_ty) = self.type_params.first() {
+            generics.push('<');
+            generics.push_str(&first_ty.to_canonical_string_with_prefix());
+            for ty in self.type_params.iter().skip(1) {
+                generics.push(',');
+                generics.push_str(&ty.to_canonical_string_with_prefix())
+            }
+            generics.push('>');
+        }
+        format!(
+            "{}::{}::{}{}",
+            self.address.to_canonical_string_with_prefix(),
+            self.module,
+            self.name,
+            generics
+        )
```

Comment:
nit (optional -- because I know this is a straight copy of the function above), here's a version of this function that minimises allocation of intermediate strings:

```rust
        let mut out = String::new();
        
        // SAFETY: Infallible
        write!(
            out, 
            "{}::{}::{}", 
            self.address.to_canonical_string_with_prefix(), 
            self.module, 
            self.name,
        ).unwrap();
        
        let mut params = self.type_params.iter();
        if let Some(first) = self.type_params.next() {
            out.push('<');
            out.push_str(&first.to_canonical_string_with_prefix());
            for param in params {
                out.push(',');
                out.push_str(&param.to_canonical_string_with_prefix());
            }
            out.push('>');
        }
        
        out
```